src/repositories/Collaborators/CollaboratorRepository.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Vertuoza\Repositories\Collaborators;
+
+use mysql_xdevapi\Warning;
+use Overblog\DataLoader\DataLoader;
+use Overblog\PromiseAdapter\PromiseAdapterInterface;
+use React\Promise\Promise;
+use Vertuoza\Repositories\BaseRepository;
+use Vertuoza\Repositories\Database\QueryBuilder;
+use Vertuoza\Repositories\Collaborators\Models\CollaboratorMapper;
+use Vertuoza\Repositories\Collaborators\Models\CollaboratorModel;
+use Vertuoza\Repositories\Collaborators\CollaboratorMutationData;
+
+use function React\Async\async;
+
+class CollaboratorRepository extends BaseRepository
+{
+  protected array $getbyIdsDL;
+  protected PromiseAdapterInterface $dataLoaderPromiseAdapter;
+
+  private const DEFAULT_PAGE_SIZE = 10;
+
+  public function __construct(
+      private QueryBuilder $database,
+      PromiseAdapterInterface $dataLoaderPromiseAdapter
+  ) {
+      parent::__construct($database, CollaboratorModel::class);
+      $this->dataLoaderPromiseAdapter = $dataLoaderPromiseAdapter;
+      $this->getbyIdsDL = [];
+  }
+
+  private function fetchByIds(string $tenantId, array $ids, int $page = 0) : Promise
```

Comment:
There should be a better way to achieve this.
This could be more readable and less complex by moving some of the default logic into BaseRepository to keep this file cleaner and have only the CRUD methods.
But keeping the logic here, it should be part of a prior (separate) refactor.